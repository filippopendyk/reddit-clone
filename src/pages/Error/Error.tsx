function Error(){
    return <h1>Error! Not found!</h1>
}

export default Error;