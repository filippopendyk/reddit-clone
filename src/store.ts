import { configureStore } from "@reduxjs/toolkit";
import filterSlice from "./features/filter/filterSlice";
import { postsSliceReducer } from "./features/posts/postsSlice";
import { fetchMostPopularCategories, topCategoriesReducer } from "./features/topCategories/topCategoriesSlice";

const store = configureStore({
    reducer: {
        topCategories: topCategoriesReducer,
        posts: postsSliceReducer,
        filter: filterSlice,
    },
});

store.dispatch(fetchMostPopularCategories());

export default store;

// Define types for usedispatch and useselector

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch;