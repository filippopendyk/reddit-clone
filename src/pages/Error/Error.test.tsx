import Error from "./Error";
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import store from "../../store";

describe('Error', () => {
    it('Renders without crashing', () => {
        const tree = renderer
        .create(
            <Provider store={store}>
                <Error/>
            </Provider>
        )
        .toJSON();
        expect(tree).toMatchSnapshot();
    })
})