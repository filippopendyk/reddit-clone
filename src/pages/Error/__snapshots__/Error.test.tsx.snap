// Vitest Snapshot v1

exports[`Error > Renders without crashing 1`] = `
<h1>
  Error! Not found!
</h1>
`;
